### PR TITLE
Remove the ticker middleware

### PR DESCRIPTION
**Pull request by fkjagodzinski**
This patch removes `hal/include/hal/ticker_api.h` which used to provide a higher layer ticker API that was mostly used to handle an event queue. This is an mbed-os specific middleware.

A couple of typedefs that are needed for the us_ticker and the lp_ticker are now placed in `hal/include/hal/ticker_types.h`.

I tried to make the commit history as explicit as I could to make reviewing simpler. Also, if anyone in the future would like to use this HAL again with the mbed-os, this should make it simpler.

### CODE REVIEW

**Comment by fkjagodzinski** [comment]
I fixed the example named `interrupt` and gave it a more meaningful name: `us_ticker_interrupt`. I also renamed the other `us_ticker` example; it is now called `us_ticker_delay`.

This is ready for review.

**Comment by MarceloSalazar** [comment]
It's looking good. Can you update `us_ticker.md` to clean-up references to events?
Perhaps we need a separate PR to update porting/testing instructions for usTicker.

**Comment by fkjagodzinski** [comment]
I updated the us_ticker porting guide to be aligned with the current state of this repo. @MarceloSalazar see the latest commit.

**Review by ithinuel — CHANGES_REQUESTED** [review]
Only the eventqueue related method should be removed.
The following should be kept:

```c
void ticker_set_handler(const ticker_data_t *const ticker, ticker_event_handler handler);
void ticker_irq_handler(const ticker_data_t *const ticker);
us_timestamp_t ticker_read_us(const ticker_data_t *const ticker);
void ticker_suspend(const ticker_data_t *const ticker);
void ticker_resume(const ticker_data_t *const ticker);
```

File: bootstrap/mbed_wait_api_no_rtos.c [hunk 1]
```diff
@@ -48,10 +45,16 @@ void _wait_us_generic(unsigned int us)
 void wait_us(int us)
 #endif
 {
-    // Generic version using full ticker, allowing for initialization, scaling and widening of timer
-    const ticker_data_t *const ticker = get_us_ticker_data();
-    const uint32_t start = ticker_read(ticker);
-    while ((ticker_read(ticker) - start) < (uint32_t)us);
+    const ticker_info_t *info = us_ticker_get_info();
```
[the remark this answers is not in the document]
Comment (fkjagodzinski) [reply bootstrap/mbed_wait_api_no_rtos.c:48]: The `ticker_read(ticker)` calls from lines 53 and 54 had to be replaced as this API has been removed. Also the while loop had to change to handle possible counter overruns; the lower layer `us_ticker_read()` does not handle overruns.

**Review by ithinuel — APPROVED** [review]

**Comment by fkjagodzinski** [comment]
> Only the eventqueue related method should be removed.

Sorry for a misleading title/description; we're actually aiming to remove the whole `ticker` layer and leave only the `us_ticker` HAL.

**Review by MarceloSalazar — APPROVED** [review]

**Pull request merged**